hivemind/optim/experimental/optimizer.py

```diff
@@ -626,9 +629,9 @@ def zero_grad(self, set_to_none: bool = False):
                 else:
                     param.grad.zero_()
 
-    def should_load_state_from_peers(self) -> bool:
+    def is_synchronized_with_peers(self) -> bool:
         """
-        If true, peer will discard local progress and attempt to download state from peers.
+        If false, peer will discard local progress and attempt to download state from peers.
```

Comment:
I'm likely to move this to a comment before loading the state from peers and rewrite the docstring

Reply:
This method specifically checks if you should download the state from peers. If that returns True, you **must call .load_state_from_peers**. If you call this method and do not load state from peers, this will result in worse synchronization guarantees.

I'd suggest **creating a new method is_synchronized_with_peers** that simply checks if local_epoch is equal or at most 1 behind the global epoch -- without changing the existing method. Furthermore,

Reply:
Created a method and made the current one private